development/main.postgressql.tf

```diff
@@ -0,0 +1,33 @@
+data "aws_ssm_parameter" "db_username" {
+  name = "/housing-finance/development/db-username"
+}
+data "aws_ssm_parameter" "db_password" {
+  name = "/housing-finance/development/db-password"
+}
+
+# HFS Postgres Master database
+module "postgres_db_master" {
+  source               = "../modules/postgres"
+  environment_name     = var.environment_name
+  db_identifier        = "housing-finance-master"
+  db_engine            = "postgres"
+  db_engine_version    = "16.1"
+  db_instance_class    = "db.t3.xlarge"
+  db_name              = "hfs-postgres-master"
+  # TODO: Replace these with a parameter group
+  db_username          = data.aws_ssm_parameter.db_username.value
+  db_password          = data.aws_ssm_parameter.db_password.value
+  vpc_id               = "vpc-0d15f152935c8716f"
+  db_allocated_storage = 1200
+  db_port              = "5432"
+  subnet_ids           = ["subnet-05ce390ba88c42bfd","subnet-0140d06fb84fdb547"]
+  storage_encrypted    = true
+  multi_az             = false
+  enabled_cloudwatch_logs_exports = ["postgresql"]
+  maintenance_window   = "tue:01:00-tue:03:00"
+  backup_window        = "08:45-09:15"
```

Comment:
I've gotta wonder about this back up window as well. Some users start their work ~7'o Clock. so this backup window cuts the incoming data stream for the day into 2 parts whilst it's still incoming.

I wonder if it's, perhaps, better to have the backup_window of hours? Like ideally before the maintenance window as well? If we make the back up say, ... at ... `23:00` o'Clock I doubt there's any incoming data happening, plus it would happen before any data altering procedures (which would give us a good rollback should a mistake in the procedure nuke some data).

Reply:
True, set the backup window to 23:00 https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/rds_cluster
In theory, multi-AZ guarantees no downtime because it switches between the DBs as it backs up but this is safer

Reply:
Yeah, though,

```tf
 multi_az             = false
 ```
 
it's probably not worth changing it to multi_az = true just for this.

Reply:
I don't mind, but even on dev?